server/src/main/java/org/opensearch/threadpool/ThreadPool.java

```diff
@@ -267,6 +270,10 @@ public ThreadPool(
             Names.REMOTE_REFRESH,
             new ScalingExecutorBuilder(Names.REMOTE_REFRESH, 1, halfProcMaxAt10, TimeValue.timeValueMinutes(5))
         );
+        builders.put(
+            Names.SEGMENT_REPLICATION,
+            new ScalingExecutorBuilder(Names.SEGMENT_REPLICATION, 1, halfProcMaxAt10, TimeValue.timeValueMinutes(5))
```

Comment:
How are we deciding on these values? Should we be trying out a few different values to see what works best with most common segment replication setups?

Reply:
+1
I think bounding `SEGMENT_REPLICATION` threadpool to max 10 threads is not correct and is problematic for cluster having high number of indices. I think having benchmark with multiple indices will come up in segment replication stats (higher replication lag) and thread pool stats.

I think for segment replication, the thread queue should not be unbounded or at least a fairly large value

Reply:
@Rishikesh1159 : Identifying ideal maximum thread count is tricky and will depend on cluster load & usage. I think we need to set this to a number sufficient to handle the busiest of traffic/ingestion pattern without bottleneck on threads in pool. One way to get more insights on this number is to run an experiment
1. Max thread count to a very high value (512?)
2. Create X indices (start with 1 ?)
3. Simulate traffic on all X indices
4. Repeat 1-3 for higher values of X

The count of max/avg thread usage will give us idea on this number. 
Please note, we need to have ingestion happening on all indices simulateneously to parallel rounds of segment replication and an increased threads usage.